src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs

```diff
@@ -124,13 +149,23 @@ public static bool IsDefinedInMongoLinqOrSystemLinq(this ISymbol symbol)
         typeSymbol is INamedTypeSymbol namedTypeSymbol &&
         s_supportedCollections.Contains(namedTypeSymbol.ConstructedFrom?.ToDisplayString());
 
+    public static bool IsSupportedStruct(this ITypeSymbol typeSymbol) =>
+        typeSymbol is INamedTypeSymbol namedTypeSymbol &&
+        s_supportedStructs.Contains(namedTypeSymbol.ConstructedFrom?.ToDisplayString());
+
     public static bool IsSupportedMongoCollectionType(this ITypeSymbol typeSymbol) =>
         typeSymbol.TypeKind == TypeKind.Class &&
         !typeSymbol.IsAnonymousType;
 
+    public static bool IsSupportedMongoAttribute(this ITypeSymbol typeSymbol) =>
+        typeSymbol.IsDefinedInMongoBson() && !s_unsupportedMongoAttributes.Contains(typeSymbol?.Name);
+
     public static bool IsString(this ITypeSymbol typeSymbol) =>
         typeSymbol?.SpecialType == SpecialType.System_String;
 
+    public static bool IsSystemType(this ITypeSymbol typeSymbol) =>
+        typeSymbol.ContainingNamespace.ToDisplayString().Equals("System");
```

Comment:
I think we need to limit, maybe with allow list, the system types we support. Both for safety and simplicity.

Reply:
DOne